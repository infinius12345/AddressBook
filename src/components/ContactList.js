import React from 'react'
import { Link } from 'react-router-dom'
import { graphql, compose } from 'react-apollo'
import { withRouter } from 'react-router-dom'
import gql from 'graphql-tag'
import {contactsLoaded} from '../actions'
import {connect} from 'react-redux'

export class Contact extends React.Component {
  state={filter:""}

  render() {
    if (this.props.contactsQuery.loading) {
      return (
        <div className="flex w-100 h-100 items-center justify-center pt7">
          <div>Loading (from {process.env.REACT_APP_GRAPHQL_ENDPOINT})</div>
        </div>
      )
    }
    let filteredContacts=[];
    if(this.props.contacts === undefined || this.props.contacts.length === 0) {
      this.props.loadContacts(this.props.contactsQuery.contacts);
    }
    else {
      filteredContacts = this.props.contacts.filter(contact => (contact.firstName + " " + contact.lastName).includes(this.state.filter))
    }
    return (
      <div>
      <input
        autoFocus
        // className="w-100 pa2 mv2 br2 b--black-20 bw1"
        className="search"
        onChange={e => this.setState({filter: e.target.value})}
        placeholder="Search using 'FirstName LastName'"
        type="text"
        value={this.state.filter}
      />
        {filteredContacts.map((contact,i) =>
        <div key={i} className="contactwrapper">
        <Link key={i}
          to={`/contact/${contact.contactId}`}
          className="list"
        >
          {contact.lastName},{contact.firstName}
        </Link>
        </div>
      )}
      </div>
    )
  }
}

export const CONTACTS_QUERY= gql`
  query contacts{
    contacts{
      lastName
      firstName
      contactId
    }
  }
`
const mapStateToProps = state => ({ });;
const mapDispatchToProps = dispatch => ({
  loadContacts: (contacts) =>
    dispatch(contactsLoaded(contacts))
})

export default compose(
  graphql(CONTACTS_QUERY, {
    name: 'contactsQuery',
  }, { skip: ({ props }) => props.contacts === undefined || props.contacts === 0}),
  withRouter,
  connect(mapStateToProps,mapDispatchToProps)
)(Contact)

