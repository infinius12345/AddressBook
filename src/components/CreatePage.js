import React from 'react'
import { withRouter } from 'react-router-dom'
import { graphql } from 'react-apollo'
import gql from 'graphql-tag'
import {connect} from 'react-redux'
import {compose} from 'react-apollo'
import ContactList from './ContactList'

export class CreatePage extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      contactId:  props.match.params.id,
      firstName: '',
      lastName: '',
      email: '',
      address: '',
      phone: '',
      set: false,
    }
    this.setContacttoState = this.setContacttoState.bind(this);

  }
  setContacttoState(contact){
    this.setState({
      firstName: contact.firstName,
      lastName: contact.lastName,
      email: contact.email,
      address: contact.address,
      phone: contact.phone,
      set:true,
    })
  }

  render() {
    if (this.props.contactQuery.loading) {
      return (
        <div className="flex w-100 h-100 items-center justify-center pt7">
          <div>Loading (from {process.env.REACT_APP_GRAPHQL_ENDPOINT})</div>
        </div>
      )
    }
    this.contact = this.props.contactQuery.contact
    if(!this.state.set && this.contact !== null){this.setContacttoState(this.contact)}

    return (
      <div id="wrapper">
        <span id="list">
          <h2 className="listtitle">All Contacts</h2>
          <ContactList contacts={this.props.contactsLoaded}/>
        </span>
        <span id="form">
        <form onSubmit={this.handlePost}>
          <h1>{this.contact ===null ? "Create": "Edit"} Contact</h1>
          <input
            autoFocus
            className="input"
            onChange={e => this.setState({ firstName: e.target.value })}
            placeholder="First Name"
            type="text"
            value={this.state.firstName}
          />
          <input
            className="input"
            cols={50}
            onChange={e => this.setState({ lastName: e.target.value })}
            placeholder="Last Name"
            rows={8}
            value={this.state.lastName}
          />
          <input
            className="input"
            cols={50}
            onChange={e => this.setState({ phone: e.target.value })}
            placeholder="Phone"
            rows={8}
            value={this.state.phone}
          />
          <input
            className="input"
            cols={50}
            onChange={e => this.setState({ email: e.target.value })}
            placeholder="Email"
            rows={8}
            type="Email"
            value={this.state.email}
          />
          <input
            className="input"
            cols={50}
            onChange={e => this.setState({ address: e.target.value })}
            placeholder="Address"
            rows={8}
            value={this.state.address}
          />
          <input
            className="input"
            type="submit"
            value={this.contact ===null ? "Create": "Edit"}
          />{' '}
        </form>
        </span>

      </div>
    )
  }

  handlePost = async e => {
    e.preventDefault();
    const { contactId,firstName,lastName,email,phone,address} = this.state;
    if(this.contact!==null) {
      await this.props.updatePersonMutation({
        variables: {contactId, firstName, lastName, email, phone, address},
      })
      this.props.history.replace(`/contact/${contactId}`)
    }
    else{
      let nextId = this.props.contactsLoaded!==0 && this.props.contactsLoaded !== undefined
        ? this.props.contactsLoaded.length + 1 : 0
      await this.props.createPersonMutation({
        variables: {contactId, firstName, lastName, email, phone, address},
      })
      this.props.history.replace(`/contact/${contactId}`)
    }

  }
}

const CREATE_PERSON_MUTATION = gql`
  mutation CreatePersonMutation($contactId: Int!,$firstName: String!, $lastName: String!,$email: String, $phone: String, $address: String) {
    createPerson(contactId:$contactId,firstName: $firstName, lastName: $lastName, email: $email, phone:$phone,address:$address) {
      contactId
      firstName
      lastName
    }
  }
`
const UPDATE_PERSON_MUTATION = gql`
  mutation UpdatePersonMutation($contactId: Int!,$firstName: String!, $lastName: String!,$email: String, $phone: String, $address: String) {
    updatePerson(contactId:$contactId,firstName: $firstName, lastName: $lastName, email: $email, phone:$phone,address:$address) {
      contactId
      firstName
      lastName
      email
      phone
      address
    }
  }
`

const CONTACT_QUERY = gql`
  query ContactQuery($id: Int!) {
    contact(contactId: $id) {
      firstName
      lastName
      email
      address
      phone
    }
  }
`

const mapStateToProps = state => ({ contactsLoaded: state.contactsLoaded});

const mapDispatchToProps = dispatch => ({
})

export default compose(
  graphql(CONTACT_QUERY, {
    name: 'contactQuery',
    options: props => ({
      variables: {
        id: props.match.params.id ? props.match.params.id : 1,
      },
    }),
  }),
  graphql(CREATE_PERSON_MUTATION, {
    name: 'createPersonMutation',
  }),
  graphql(UPDATE_PERSON_MUTATION, {
    name: 'updatePersonMutation',
  }),
  withRouter,
  connect(mapStateToProps,mapDispatchToProps)
)(CreatePage)
