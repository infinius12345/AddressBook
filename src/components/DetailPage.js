import React from 'react'
import {graphql, compose} from 'react-apollo'
import {Link, withRouter} from 'react-router-dom'
import gql from 'graphql-tag'
import ContactList from './ContactList'
import {connect} from 'react-redux'
import {contactsLoaded} from '../actions/index'

export class DetailPage extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      filter: '',
    }
  }

  render() {
    if (this.props.contactquery.loading) {
      return (
        <div className="flex w-100 h-100 items-center justify-center pt7">
          <div>Loading (from {process.env.REACT_APP_GRAPHQL_ENDPOINT})</div>
        </div>
      )
    }
    //console.log(this.props);
    let {contact} = this.props.contactquery

    if(contact===null){
      contact={};
      contact.firstName="No";
      contact.lastName = "found";
      contact.email = '';
      contact.address = '';
      contact.phone = '';
    }

    return (
      <React.Fragment>
        <div id="wrapper">
          <span id="list">
            <h2 className="listtitle">All Contacts</h2>
            <ContactList contacts={this.props.contactsLoaded}/>
          </span>

          <span id="detail">
            <div id="padding"><h1 className="heading">{contact.firstName} {contact.lastName}</h1>
              <Link className="link"
            to={`/edit/${contact.contactId}`}
              >Edit
            </Link></div>

            <table id="table">
              <tr id="row"><td id="column1">Phone:</td><td>{contact.phone}</td></tr>
              <tr id="row"><td id="column1">Email:</td><td>{contact.email}</td></tr>
              <tr id="row"><td id="column1">Address:</td><td>{contact.address}</td></tr>
            </table>


          </span>
        </div>


      </React.Fragment>
    )
  }

  deletePost = async id => {
    await this.props.deletePost({
      variables: {id},
    })
    this.props.history.replace('/')
  }

  publishDraft = async id => {
    await this.props.publishDraft({
      variables: {id},
    })
    this.props.history.replace('/')
  }
}

export const CONTACT_QUERY = gql`
  query contactquery($id: Int!) {
    contact(contactId: $id) {
      contactId
      firstName
      lastName
      email
      phone
      address
    }
  }
`

const mapStateToProps = state => ({contactsLoaded: state.contactsLoaded});

const mapDispatchToProps = dispatch => ({
  loadContacts: (contacts) =>
    dispatch(contactsLoaded(contacts)),
})

export default compose(
  graphql(CONTACT_QUERY, {
    name: 'contactquery',
    options: props => ({
      variables: {
        id: props.match.params.id ? props.match.params.id : 1,
      },
    }),
  }),
  withRouter,
  connect(mapStateToProps, mapDispatchToProps)
)(DetailPage)
