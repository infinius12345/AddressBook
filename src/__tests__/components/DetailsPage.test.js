import React from 'react';
import { shallow, configure } from 'enzyme';
import {jest} from 'jest';
import Adapter from 'enzyme-adapter-react-16';
import {DetailPage} from '../../components/DetailPage';

configure({ adapter: new Adapter() });

const contactQueryFalse = {
  loading: false,
  contact:
    {
      firstName: "Hello",
      lastName: "No",
      email:"adsf@gmail.com",
      phone:"21312",
      address:"dsafdsaf",
    }
}

const match={
  params:{
    id:1
  }
};

test('ContactList renders correctly after Query', () => {
  const component = shallow(<DetailPage
    contactquery={contactQueryFalse}
    match={match}
  />);
  expect(component).toMatchSnapshot();
});
