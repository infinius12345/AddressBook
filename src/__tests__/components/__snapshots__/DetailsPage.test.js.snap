// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList renders correctly after Query 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DetailPage
    contactquery={
      Object {
        "contact": Object {
          "address": "dsafdsaf",
          "email": "adsf@gmail.com",
          "firstName": "Hello",
          "lastName": "No",
          "phone": "21312",
        },
        "loading": false,
      }
    }
    match={
      Object {
        "params": Object {
          "id": 1,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        id="wrapper"
      >
        <span
          id="list"
        >
          <h2
            className="listtitle"
          >
            All Contacts
          </h2>
          <Apollo(withRouter(Connect(Contact)))
            contacts={undefined}
          />
        </span>
        <span
          id="detail"
        >
          <div
            id="padding"
          >
            <h1
              className="heading"
            >
              Hello
               
              No
            </h1>
            <Link
              className="link"
              replace={false}
              to="/edit/undefined"
            >
              Edit
            </Link>
          </div>
          <table
            id="table"
          >
            <tr
              id="row"
            >
              <td
                id="column1"
              >
                Phone:
              </td>
              <td>
                21312
              </td>
            </tr>
            <tr
              id="row"
            >
              <td
                id="column1"
              >
                Email:
              </td>
              <td>
                adsf@gmail.com
              </td>
            </tr>
            <tr
              id="row"
            >
              <td
                id="column1"
              >
                Address:
              </td>
              <td>
                dsafdsaf
              </td>
            </tr>
          </table>
        </span>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            id="list"
          >
            <h2
              className="listtitle"
            >
              All Contacts
            </h2>
            <Apollo(withRouter(Connect(Contact)))
              contacts={undefined}
            />
          </span>,
          <span
            id="detail"
          >
            <div
              id="padding"
            >
              <h1
                className="heading"
              >
                Hello
                 
                No
              </h1>
              <Link
                className="link"
                replace={false}
                to="/edit/undefined"
              >
                Edit
              </Link>
            </div>
            <table
              id="table"
            >
              <tr
                id="row"
              >
                <td
                  id="column1"
                >
                  Phone:
                </td>
                <td>
                  21312
                </td>
              </tr>
              <tr
                id="row"
              >
                <td
                  id="column1"
                >
                  Email:
                </td>
                <td>
                  adsf@gmail.com
                </td>
              </tr>
              <tr
                id="row"
              >
                <td
                  id="column1"
                >
                  Address:
                </td>
                <td>
                  dsafdsaf
                </td>
              </tr>
            </table>
          </span>,
        ],
        "id": "wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="listtitle"
              >
                All Contacts
              </h2>,
              <Apollo(withRouter(Connect(Contact)))
                contacts={undefined}
              />,
            ],
            "id": "list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "All Contacts",
                "className": "listtitle",
              },
              "ref": null,
              "rendered": "All Contacts",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "contacts": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                id="padding"
              >
                <h1
                  className="heading"
                >
                  Hello
                   
                  No
                </h1>
                <Link
                  className="link"
                  replace={false}
                  to="/edit/undefined"
                >
                  Edit
                </Link>
              </div>,
              <table
                id="table"
              >
                <tr
                  id="row"
                >
                  <td
                    id="column1"
                  >
                    Phone:
                  </td>
                  <td>
                    21312
                  </td>
                </tr>
                <tr
                  id="row"
                >
                  <td
                    id="column1"
                  >
                    Email:
                  </td>
                  <td>
                    adsf@gmail.com
                  </td>
                </tr>
                <tr
                  id="row"
                >
                  <td
                    id="column1"
                  >
                    Address:
                  </td>
                  <td>
                    dsafdsaf
                  </td>
                </tr>
              </table>,
            ],
            "id": "detail",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="heading"
                  >
                    Hello
                     
                    No
                  </h1>,
                  <Link
                    className="link"
                    replace={false}
                    to="/edit/undefined"
                  >
                    Edit
                  </Link>,
                ],
                "id": "padding",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Hello",
                      " ",
                      "No",
                    ],
                    "className": "heading",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Hello",
                    " ",
                    "No",
                  ],
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Edit",
                    "className": "link",
                    "replace": false,
                    "to": "/edit/undefined",
                  },
                  "ref": null,
                  "rendered": "Edit",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <tr
                    id="row"
                  >
                    <td
                      id="column1"
                    >
                      Phone:
                    </td>
                    <td>
                      21312
                    </td>
                  </tr>,
                  <tr
                    id="row"
                  >
                    <td
                      id="column1"
                    >
                      Email:
                    </td>
                    <td>
                      adsf@gmail.com
                    </td>
                  </tr>,
                  <tr
                    id="row"
                  >
                    <td
                      id="column1"
                    >
                      Address:
                    </td>
                    <td>
                      dsafdsaf
                    </td>
                  </tr>,
                ],
                "id": "table",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <td
                        id="column1"
                      >
                        Phone:
                      </td>,
                      <td>
                        21312
                      </td>,
                    ],
                    "id": "row",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Phone:",
                        "id": "column1",
                      },
                      "ref": null,
                      "rendered": "Phone:",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "21312",
                      },
                      "ref": null,
                      "rendered": "21312",
                      "type": "td",
                    },
                  ],
                  "type": "tr",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <td
                        id="column1"
                      >
                        Email:
                      </td>,
                      <td>
                        adsf@gmail.com
                      </td>,
                    ],
                    "id": "row",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Email:",
                        "id": "column1",
                      },
                      "ref": null,
                      "rendered": "Email:",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "adsf@gmail.com",
                      },
                      "ref": null,
                      "rendered": "adsf@gmail.com",
                      "type": "td",
                    },
                  ],
                  "type": "tr",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <td
                        id="column1"
                      >
                        Address:
                      </td>,
                      <td>
                        dsafdsaf
                      </td>,
                    ],
                    "id": "row",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Address:",
                        "id": "column1",
                      },
                      "ref": null,
                      "rendered": "Address:",
                      "type": "td",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "dsafdsaf",
                      },
                      "ref": null,
                      "rendered": "dsafdsaf",
                      "type": "td",
                    },
                  ],
                  "type": "tr",
                },
              ],
              "type": "table",
            },
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          id="wrapper"
        >
          <span
            id="list"
          >
            <h2
              className="listtitle"
            >
              All Contacts
            </h2>
            <Apollo(withRouter(Connect(Contact)))
              contacts={undefined}
            />
          </span>
          <span
            id="detail"
          >
            <div
              id="padding"
            >
              <h1
                className="heading"
              >
                Hello
                 
                No
              </h1>
              <Link
                className="link"
                replace={false}
                to="/edit/undefined"
              >
                Edit
              </Link>
            </div>
            <table
              id="table"
            >
              <tr
                id="row"
              >
                <td
                  id="column1"
                >
                  Phone:
                </td>
                <td>
                  21312
                </td>
              </tr>
              <tr
                id="row"
              >
                <td
                  id="column1"
                >
                  Email:
                </td>
                <td>
                  adsf@gmail.com
                </td>
              </tr>
              <tr
                id="row"
              >
                <td
                  id="column1"
                >
                  Address:
                </td>
                <td>
                  dsafdsaf
                </td>
              </tr>
            </table>
          </span>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              id="list"
            >
              <h2
                className="listtitle"
              >
                All Contacts
              </h2>
              <Apollo(withRouter(Connect(Contact)))
                contacts={undefined}
              />
            </span>,
            <span
              id="detail"
            >
              <div
                id="padding"
              >
                <h1
                  className="heading"
                >
                  Hello
                   
                  No
                </h1>
                <Link
                  className="link"
                  replace={false}
                  to="/edit/undefined"
                >
                  Edit
                </Link>
              </div>
              <table
                id="table"
              >
                <tr
                  id="row"
                >
                  <td
                    id="column1"
                  >
                    Phone:
                  </td>
                  <td>
                    21312
                  </td>
                </tr>
                <tr
                  id="row"
                >
                  <td
                    id="column1"
                  >
                    Email:
                  </td>
                  <td>
                    adsf@gmail.com
                  </td>
                </tr>
                <tr
                  id="row"
                >
                  <td
                    id="column1"
                  >
                    Address:
                  </td>
                  <td>
                    dsafdsaf
                  </td>
                </tr>
              </table>
            </span>,
          ],
          "id": "wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="listtitle"
                >
                  All Contacts
                </h2>,
                <Apollo(withRouter(Connect(Contact)))
                  contacts={undefined}
                />,
              ],
              "id": "list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "All Contacts",
                  "className": "listtitle",
                },
                "ref": null,
                "rendered": "All Contacts",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "contacts": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  id="padding"
                >
                  <h1
                    className="heading"
                  >
                    Hello
                     
                    No
                  </h1>
                  <Link
                    className="link"
                    replace={false}
                    to="/edit/undefined"
                  >
                    Edit
                  </Link>
                </div>,
                <table
                  id="table"
                >
                  <tr
                    id="row"
                  >
                    <td
                      id="column1"
                    >
                      Phone:
                    </td>
                    <td>
                      21312
                    </td>
                  </tr>
                  <tr
                    id="row"
                  >
                    <td
                      id="column1"
                    >
                      Email:
                    </td>
                    <td>
                      adsf@gmail.com
                    </td>
                  </tr>
                  <tr
                    id="row"
                  >
                    <td
                      id="column1"
                    >
                      Address:
                    </td>
                    <td>
                      dsafdsaf
                    </td>
                  </tr>
                </table>,
              ],
              "id": "detail",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h1
                      className="heading"
                    >
                      Hello
                       
                      No
                    </h1>,
                    <Link
                      className="link"
                      replace={false}
                      to="/edit/undefined"
                    >
                      Edit
                    </Link>,
                  ],
                  "id": "padding",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "Hello",
                        " ",
                        "No",
                      ],
                      "className": "heading",
                    },
                    "ref": null,
                    "rendered": Array [
                      "Hello",
                      " ",
                      "No",
                    ],
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Edit",
                      "className": "link",
                      "replace": false,
                      "to": "/edit/undefined",
                    },
                    "ref": null,
                    "rendered": "Edit",
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <tr
                      id="row"
                    >
                      <td
                        id="column1"
                      >
                        Phone:
                      </td>
                      <td>
                        21312
                      </td>
                    </tr>,
                    <tr
                      id="row"
                    >
                      <td
                        id="column1"
                      >
                        Email:
                      </td>
                      <td>
                        adsf@gmail.com
                      </td>
                    </tr>,
                    <tr
                      id="row"
                    >
                      <td
                        id="column1"
                      >
                        Address:
                      </td>
                      <td>
                        dsafdsaf
                      </td>
                    </tr>,
                  ],
                  "id": "table",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <td
                          id="column1"
                        >
                          Phone:
                        </td>,
                        <td>
                          21312
                        </td>,
                      ],
                      "id": "row",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Phone:",
                          "id": "column1",
                        },
                        "ref": null,
                        "rendered": "Phone:",
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "21312",
                        },
                        "ref": null,
                        "rendered": "21312",
                        "type": "td",
                      },
                    ],
                    "type": "tr",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <td
                          id="column1"
                        >
                          Email:
                        </td>,
                        <td>
                          adsf@gmail.com
                        </td>,
                      ],
                      "id": "row",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Email:",
                          "id": "column1",
                        },
                        "ref": null,
                        "rendered": "Email:",
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "adsf@gmail.com",
                        },
                        "ref": null,
                        "rendered": "adsf@gmail.com",
                        "type": "td",
                      },
                    ],
                    "type": "tr",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <td
                          id="column1"
                        >
                          Address:
                        </td>,
                        <td>
                          dsafdsaf
                        </td>,
                      ],
                      "id": "row",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "Address:",
                          "id": "column1",
                        },
                        "ref": null,
                        "rendered": "Address:",
                        "type": "td",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "dsafdsaf",
                        },
                        "ref": null,
                        "rendered": "dsafdsaf",
                        "type": "td",
                      },
                    ],
                    "type": "tr",
                  },
                ],
                "type": "table",
              },
            ],
            "type": "span",
          },
        ],
        "type": "div",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
